lib/std/fs.zig

```diff
@@ -724,6 +744,19 @@ pub const Dir = struct {
             try std.event.Loop.instance.?.openatZ(self.fd, sub_path_c, os_flags, flags.mode)
         else
             try os.openatC(self.fd, sub_path_c, os_flags, flags.mode);
+
+        if (flags.lock) {
```

Comment:
The *BSD systems have two interesting flags for `open`/`openat` named `O_EXLOCK` and `O_SHLOCK`, the `fcntl` trick and the wonky exclusive locks are a Linux-only problem :P

If the underlying filesystem doesn't support locking `EOPNOTSUPP` is returned.

Reply:
This seems like a good enhancement to the API, but I think I'll leave this for a later unless there is a relatively straightforward way to check if the system supports those flags.

Reply:
> unless there is a relatively straightforward way to check if the system supports those flags

You just have to handle `EOPNOTSUPP`, if the filesystem doesn't support file locks then using `fcntl` will fail as well.